velox/connectors/hive/HiveConnector.cpp

```diff
@@ -81,28 +91,67 @@ std::string HiveTableHandle::toString() const {
 
 HiveDataSink::HiveDataSink(
     std::shared_ptr<const RowType> inputType,
-    const std::string& filePath,
-    velox::memory::MemoryPool* memoryPool)
-    : inputType_(inputType) {
+    std::shared_ptr<const HiveInsertTableHandle> insertTableHandle,
+    const ConnectorQueryCtx* FOLLY_NONNULL connectorQueryCtx,
+    std::shared_ptr<WriteProtocol> writeProtocol)
+    : inputType_(inputType),
+      insertTableHandle_(insertTableHandle),
+      connectorQueryCtx_(connectorQueryCtx),
+      writeProtocol_(writeProtocol),
+      writerParameters_{},
+      writers_{} {
+  if (connectorQueryCtx_->queryConfig()->createEmptyFiles() &&
+      !insertTableHandle_->isPartitioned()) {
+    LOG(INFO) << "Enforce creating empty files\n";
+    writers_.emplace_back(createWriter());
```

Comment:
Do we ever create >1 writer? If so, consider changing the type from std::vector and just a writer.